### PR TITLE
Upgrade dependencies in dockerfile

### PR DESCRIPTION
**Pull request by Jacobh2**
## Update dependencies in dockerfiles

Updating the dependencies in dockerfiles

## Relevant issues

https://github.com/BerriAI/litellm/issues/5630

## Type

🐛 Dependency Upgrade

## Changes

- Lock the dependency versions for more consistent builds and upgrade debian packages.
- Update the image packages
- Change alpine image to use `apk` instead of `apt-get`.

### CODE REVIEW

File: Dockerfile
Comment (superpoussin22) [inline Dockerfile:1]: update done twice

File: Dockerfile [hunk 1]
```diff
@@ -41,7 +41,7 @@ RUN pip install redisvl==0.0.7 --no-deps
 # ensure pyjwt is used, not jwt
 RUN pip uninstall jwt -y
 RUN pip uninstall PyJWT -y
-RUN pip install PyJWT --no-cache-dir
+RUN pip install PyJWT==2.9.0 --no-cache-dir
```
Comment (ishaan-jaff) [inline Dockerfile:44]: @Jacobh2 this would be out of sync with requirements.txt which sets it to 2.8.0 

can you also bump requirements.txt

**Review by ishaan-jaff — CHANGES_REQUESTED** [review]
change requested

File: Dockerfile (hunk 1 above)
Comment (Jacobh2) [reply Dockerfile:44]: Done! 👍

**Review by ishaan-jaff — APPROVED** [review]
thanks for this change @Jacobh2 !

**Pull request merged**